ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoice.java

```diff
@@ -272,6 +272,14 @@ private BlockImportResult importBlockAndState(
         transaction.setProposerBoostRoot(block.getRoot());
       }
     }
+
+    if (payloadResult.getStatus() == ExecutionPayloadStatus.VALID) {
+      UInt64 latestValidFinalizedSlot = transaction.getLatestFinalized().getSlot();
+      if (latestValidFinalizedSlot.compareTo(transaction.getLatestValidFinalizedSlot()) != 0) {
```

Comment:
I think this is fine as is but we shouldn't ever let the latest valid finalised slot go backwards so I'd be tempted to use:

```suggestion
      if (latestValidFinalizedSlot.isGreaterThan(transaction.getLatestValidFinalizedSlot())) {
```

mostly just for clarity but also a little bit of defensiveness to ensure it never goes backwards.

Reply:
i'm always pro defensiveness, but in this case i'd like to understand better the scenario.

AFAIK finalised anchor should never go backward, right?
Assuming that it does, why you think that we should not move back the slot accordingly? In other words, which side effect we want to avoid?

Reply:
I'm thinking of a scenario where we are concurrently performing the state transition for two blocks, one of which updates the finalised checkpoint (A) and the other doesn't (B).  Since we've done the state transition prior to jumping to the fork choice thread it's possible that block B completes validation against the old finalised checkpoint but we actually add block A to the store first, updating the checkpoint.  B will then be imported - it's no longer relevant and will be ignored by fork choice and eventually pruned but we need to make sure it doesn't lower the latest valid finalised slot when it imports.